File: packages/dart/lib/src/models/county.dart
/// Represents a county in Kenya.
///
/// Contains the county's identifying information, administrative region,
/// geographical area, population, and postal code.
class County {
  /// Creates a [County] with the given administrative and demographic data.
  const new({
    required this.code,
    required this.name,
    required this.capital,
    required this.areaKm2,
    required this.population2019,
    required this.region,
    required this.postalCode,
  });

  /// Creates a [County] from a map containing county data.
  ///
  /// The map is expected to contain the following keys:
  /// - `code`
  /// - `name`
  /// - `capital`
  /// - `area_km2`
  /// - `population_2019`
  /// - `region`
  /// - `postal_code`
  factory fromMap(Map<String, dynamic> map) => County(
    code: map['code'] as String,
    name: map['name'] as String,
    capital: map['capital'] as String,
    areaKm2: (map['area_km2'] as num).toDouble(),
    population2019: map['population_2019'] as int,
    region: map['region'] as String,
    postalCode: map['postal_code'] as String,
  );

  /// The unique administrative code of the county.
  final String code;

  /// The name of the county.
  final String name;

  /// The capital or administrative headquarters of the county.
  final String capital;

  /// The geographical area of the county in square kilometres.
  final double areaKm2;

  /// The county's population according to the 2019 Kenya Population and
  /// Housing Census.
  final int population2019;

  /// The geographical or administrative region associated with the county.
  final String region;

  /// The postal code associated with the county.
  final String postalCode;
}

File: packages/php/src/County.php
<?php

declare(strict_types=1);

namespace KenyaLocations;

/**
 * A Kenyan county with administrative and demographic metadata.
 */
final readonly class County
{
    /**
     * @param array<string, mixed> $data
     */
    public static function fromArray(array $data): self
    {
        return new self(
            code: (string) $data['code'],
            name: (string) $data['name'],
            capital: (string) $data['capital'],
            areaKm2: (float) $data['area_km2'],
            population2019: (int) $data['population_2019'],
            region: (string) $data['region'],
            postalCode: (string) $data['postal_code'],
        );
    }

    public function __construct(
        public string $code,
        public string $name,
        public string $capital,
        public float $areaKm2,
        public int $population2019,
        public string $region,
        public string $postalCode,
    ) {
    }
}
